### PR TITLE
DOCSP-41853 Updates mongosync documentation to support 1.8 release

### PR DESCRIPTION
**Pull request by jwilson-mdb**
# Description
Makes requested documentation changes to support the release of `mongosync` 1.8.0

Note to reviewers: The writing was internally reviewed in #352. This PR moves the writing out of release notes and into the documentation. 

# JIRA
https://jira.mongodb.org/browse/DOCSP-41853

# Staging
[/start](https://preview-mongodbjwilsonmdb.gatsbyjs.io/cluster-sync/DOCSP-41853/reference/api/start/#required-indexes)
[/progress - totalEventsApplied added to response](https://preview-mongodbjwilsonmdb.gatsbyjs.io/cluster-sync/DOCSP-41853/reference/api/progress/#successful-response)
[/progress - adds new behaviors to Behavior section](https://preview-mongodbjwilsonmdb.gatsbyjs.io/cluster-sync/DOCSP-41853/reference/api/progress/#behavior)
[telemetry - moves "Hostnames" from untracked to tracked](https://preview-mongodbjwilsonmdb.gatsbyjs.io/cluster-sync/DOCSP-41853/reference/telemetry/#data-collection)

# Build log
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=66a3ed8b26eba7955278f5b6

### CODE REVIEW

**Review by jmd-mongo — APPROVED** [review]
Thanks for these helpful updates, @jwilson-mdb! This issue LGTM.

Thanks,
Joe

File: source/reference/api/progress.txt [hunk 1]
```diff
@@ -64,7 +64,25 @@ Behavior
 
 - The endpoint does not auto-refresh. To get updated status, call the
   ``progress`` endpoint again.
-  
+
+- Calls to ``/progress`` before ``mongosync`` reaches the collection
+  copy phase return 0 for ``estimatedCopiedBytes`` and 1 for
```
Comment (jtcovan) [inline source/reference/api/progress.txt:69]:
```suggestion
  copy phase returns 0 for ``estimatedCopiedBytes`` and 1 for 
```

File: source/reference/api/progress.txt [hunk 2]
```diff
@@ -64,7 +64,25 @@ Behavior
...
-  
+
+- Calls to ``/progress`` before ``mongosync`` reaches the collection
+  copy phase return 0 for ``estimatedCopiedBytes`` and 1 for 
+  ``estimatedTotalBytes``.
+
+- ``estimatedTotalBytes`` may change throughout the collection copy 
+  phase if documents are inserted or deleted on the source cluster.
+
+- ``estimatedCopiedBytes`` is never be greater than
```
Comment (jtcovan) [inline source/reference/api/progress.txt:75]:
```suggestion
- ``estimatedCopiedBytes`` is never greater than 
```

**Review by jtcovan — CHANGES_REQUESTED** [review]
Mostly looks good to me! One other change though,

Can we add to the `Successful Response` on the [progress page](https://www.mongodb.com/docs/cluster-to-cluster-sync/current/reference/api/progress/#successful-response) that there will be a top-level field `success` that is `true`? Either in the description or by adding the `success` field to the table and moving all the existing fields as part of the `progress` object field.

And to the [error response](https://www.mongodb.com/docs/cluster-to-cluster-sync/current/reference/api/progress/#error-response) can we add the same `success`, `error`, and `errorDescription` fields as described in this API page [page](https://www.mongodb.com/docs/cluster-to-cluster-sync/current/reference/api/start/#response)?

File: source/reference/api/progress.txt (hunk 1 above)
Comment (jwilson-mdb) [reply source/reference/api/progress.txt:69]: Here, I believe "return" is correct as "Calls" is a plural subject.
`[API] Calls return...` vs `[an API] Call returns`

**Comment by jwilson-mdb** [comment]
@jtcovan I think that the `/progress` changes you are requesting are already present and were merged in #295.
Staging link to [Successful Response](https://www.mongodb.com/docs/cluster-to-cluster-sync/v1.8/reference/api/progress/#successful-response:~:text=The%20response%20object%20contains%202%20top%2Dlevel%20fields%2C%20success%20and%20progress.)
Staging link to [/progress Error Response table](https://preview-mongodbjwilsonmdb.gatsbyjs.io/cluster-sync/DOCSP-41853/reference/api/progress/#error-response:~:text=Description-,success,true%20if%20the%20command%20succeeds%20and%20false%20if%20the%20command%20fails.,-error)

If I am misunderstanding this, or if you'd like to tweak any of the existing wording please let me know and I'll make the changes.

File: source/reference/api/progress.txt (hunk 1 above)
Comment (jtcovan) [reply source/reference/api/progress.txt:69]: Agree, unable to resolve this comment

**Review by jtcovan — APPROVED** [review]
You're right, my bad. LGTM, thanks

**Pull request merged**